diff --git a/drivers/media/platform/aspeed-video.c b/drivers/media/platform/aspeed-video.c
index 1bb863b..b783f18 100644
--- a/drivers/media/platform/aspeed-video.c
+++ b/drivers/media/platform/aspeed-video.c
@@ -748,6 +748,10 @@ static void aspeed_video_get_resolution(struct aspeed_video *video)
 					    VE_MAX_SRC_BUFFER_SIZE)) {
 			dev_err(video->dev,
 				"Failed to allocate source buffers\n");
+			dev_err(video->dev, "srcs[0].size %d; srcs[0].dma %d, srcs[0].virt 0x%p\n", 
+				video->srcs[0].size,
+				video->srcs[0].dma,
+				video->srcs[0].virt);
 			return;
 		}
 	}
@@ -897,7 +901,7 @@ static void aspeed_video_set_resolution(struct aspeed_video *video)
 	return;
 
 err_mem:
-	dev_err(video->dev, "Failed to allocate source buffers\n");
+	dev_err(video->dev, "Failed to allocate source buffers (set_resolution)\n");
 
 	if (video->srcs[0].size)
 		aspeed_video_free_buf(video, &video->srcs[0]);
